crates/rooch-framework/sources/auth_validator/bitcoin_validator.move

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) RoochNetwork
+// SPDX-License-Identifier: Apache-2.0
+
+/// This module implements Ethereum validator with the ECDSA recoverable signature over Secp256k1.
+module rooch_framework::bitcoin_validator {
+
+    use std::vector;
+    use std::option::{Self, Option};
+    use std::signer;
+    use moveos_std::context::{Self, Context};
+    use rooch_framework::account_authentication;
+    use rooch_framework::ecdsa_k1_recoverable;
+    use rooch_framework::auth_validator;
+    use rooch_framework::ethereum_address::{Self, ETHAddress};
+
+    /// there defines auth validator id for each blockchain
+    const BITCOIN_AUTH_VALIDATOR_ID: u64 = 2;
+
+    // error code
+    const ErrorInvalidPublicKeyLength: u64 = 1;
+
+    struct BitcoinValidator has store, drop {}
+
+    public fun auth_validator_id(): u64 {
+        BITCOIN_AUTH_VALIDATOR_ID
+    }
+
+    public entry fun rotate_authentication_key_entry(
+        ctx: &mut Context,
+        account: &signer,
+        public_key: vector<u8>
+    ) {
+        // compare newly passed public key with Ethereum public key length to ensure it's compatible
+        assert!(
+            vector::length(&public_key) == ecdsa_k1_recoverable::public_key_length(),
+            ErrorInvalidPublicKeyLength
+        );
+
+        // User can rotate the authentication key arbitrarily, so we do not need to check the new public key with the account address.
+        let authentication_key = public_key_to_authentication_key(public_key);
+        let account_addr = signer::address_of(account);
+        rotate_authentication_key(ctx, account_addr, authentication_key);
+    }
+
+    fun rotate_authentication_key(ctx: &mut Context, account_addr: address, authentication_key: vector<u8>) {
+        account_authentication::rotate_authentication_key<BitcoinValidator>(ctx, account_addr, authentication_key);
+    }
+
+    public entry fun remove_authentication_key_entry(ctx: &mut Context, account: &signer) {
+        account_authentication::remove_authentication_key<BitcoinValidator>(ctx, signer::address_of(account));
+    }
+
```

Comment:
Bitcoin auth validator does not need to support the rotate authentication key.

Reply:
I removed all the code associated with the authentication key